prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,298 @@
+
+    # weekday first
+    trip_num = 1
+    start_point = 0
+    end_point = 0
+
+    for i in range(l):
+        if data_day[i] == 2:  # 2 is weekday
+            if (
+                i > 0
+                and newdata.iloc[i, newdata.columns.get_loc("sample vehicle number")]
+                == newdata.iloc[i - 1, newdata.columns.get_loc("sample vehicle number")]
+            ):
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = (
+                    newdata.iloc[
+                        i - 1, newdata.columns.get_loc("trip end battery charge")
+                    ]
+                    + newdata.iloc[
+                        i - 1, newdata.columns.get_loc("charging consumption")
+                    ]
+                )
+                trip_num += 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            else:
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = kwh
+                trip_num = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            if (
+                newdata.iloc[i, newdata.columns.get_loc("total vehicle miles traveled")]
+                < veh_range * const.ER
+            ):
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")] = (
+                    newdata.iloc[
+                        i, newdata.columns.get_loc("trip start battery charge")
+                    ]
+                    - newdata.iloc[i, newdata.columns.get_loc("Miles traveled")]
+                    * kwhmi
+                    * const.ER
+                )
+            else:
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 0
+
+            trip_data = newdata.iloc[i]
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging power")
+            ] = charging.get_charging_power(
+                power,
+                trip_strategy,
+                location_strategy,
+                kwh,
+                trip_data,
+            )
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging time")
+            ] = charging.get_charging_time(
+                newdata.iloc[i, newdata.columns.get_loc("charging power")],
+                kwh,
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")],
+                const.charging_efficiency,
+                trip_data,
+            )
+
+            newdata.iloc[i, newdata.columns.get_loc("charging consumption")] = (
+                newdata.iloc[i, newdata.columns.get_loc("charging power")]
+                * newdata.iloc[i, newdata.columns.get_loc("charging time")]
+                * const.charging_efficiency
+            )
```

Comment:
Same comment as above: this can be down for all rows at once by operating on entire columns.